src/EIP7702Proxy.sol

```diff
@@ -186,20 +195,18 @@ contract EIP7702Proxy is Proxy {
         return _ERC1271_FAIL_VALUE;
     }
 
-    /// @notice Returns the implementation address, falling back to the initial implementation if the ERC-1967 implementation slot is not set
+    /// @notice Returns the ERC-1967 implementation address
     ///
     /// @return implementation The implementation address for this EOA
     function _implementation() internal view override returns (address) {
-        address implementation = ERC1967Utils.getImplementation();
-        return
-            implementation == address(0)
-                ? INITIAL_IMPLEMENTATION
-                : implementation;
+        return ERC1967Utils.getImplementation();
     }
 
     /// @inheritdoc Proxy
-    /// @dev Guards a specified initializer function from being called directly
+    /// @dev Guards a specified initializer function from being called directly. Reverts if the proxy is not initialized.
     function _fallback() internal override {
+        if (_implementation() == address(0)) revert Uninitialized();
```

Comment:
again, I think we could actually still get away with having a default fallback implementation and not blocking these calls, but I'd be down to try to nix the ick

Reply:
connecting the dots on other comment regarding not inheriting `Receiver`, I was intuiting we would delegatecall the immutable `RECEIVER` here if implementation is empty.

Reply:
Effectively, initialImplementation is the receiver actually. We force users to set the implementation intentionally and have it pass a validator before they can use functionality. We've only baked in backwards compatibility via 1271 and token receivers while in the uninitialized state